includes/process-download.php

```diff
@@ -838,3 +850,30 @@ function edd_set_requested_file_scheme( $requested_file, $download_files, $file_
 
 }
 add_filter( 'edd_requested_file', 'edd_set_requested_file_scheme', 10, 3 );
+
+/**
+ * Perform a head request on file URLs before attempting to download to check if they are accessible.
+ *
+ * @since  2.6.14
+ * @param  string $requested_file The Requested File
+ * @param  array  $args           Arguments
+ * @param  string $method         The download mehtod being sed
+ * @return void
+ */
+function edd_check_file_url_head( $requested_file, $args, $method ) {
+
+	// If this is a file URL (not a path), perform a head request to determine if it's valid
+	if( filter_var( $requested_file, FILTER_VALIDATE_URL ) ) {
+
+		$request = wp_remote_head( $requested_file );
+
+		if( is_wp_error( $request) ) {
```

Comment:
I was having a hard time triggering a 404 error code. It was strange, even files that were clearly 404'ing were returning a 200 code with wp_remote_head(). No idea why.

Checking for a 404 code does seem like a good idea.